### DISCUSSION TITLE
Is it possible to offer an option for gtk-mac-integration?

### DISCUSSION THREAD

**Discussion by easior** [Homebrew/discussions#1623] [Tap maintenance and Homebrew development]
The formula gtk-mac-integration is useful in several aspects: to build an integrated menu with Mac OS X, or to install RAD tool Glade, etc. However, The installation of gtk-mac-integration would depend both on gtk+2 and gtk+3, it is inconvenient. Could the dependence on both GTK branchs be reduced to only one? I hope, it could be offered two option, e.g. gtk+2, gtk+3, which give the user a choice to install a proper GTK dependence.

**Comment by SMillerDev** [comment]
In a tap this can be done very easily doing `depends_on "y" => :optional`.
In homebrew core it won't be possible.

**Reply by easior** [reply]
Thank you for your suggestion, @SMillerDev 

I hope the maintainer of gtk-mac-integration would adopt this suggestion.

**Reply by gromgit** [reply]
You missed the last sentence:
> In homebrew core it won't be possible.

If you need this to happen, _you_ need to create your own tap and copy the `gtk-mac-integration` formula into it, then you can make it do whatever you want: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

**Reply by easior** [reply]
Thank you for your comment @gromgit.
Actually, I have made my own gtk-mac-integration formula in my tap. There is still one thing I can't figure out:
how to let the options work as arbitrary one over two choices or both two?
In my situation, gtk+ is optional now, but gtk+3 couldn't be, because my modification is as follows
depends_on "gtk+" => :optional

**Reply by gromgit** [reply]
> how to let the options work as arbitrary one over two choices or both two?

Please read https://docs.brew.sh/Formula-Cookbook#specifying-other-formulae-as-dependencies carefully. In particular:
> * `:optional` generates an implicit `with-foo` option for the formula. This means that, given `depends_on "foo" => :optional`, the user must pass `--with-foo` in order to use the dependency.

If you don't tag `gtk+3` as `:optional` too, it becomes a _mandatory_ dependency.

**Reply by carlocab** [reply]
You could tag `gtk+3` as `:recommended`.

**Reply by gromgit** [reply]
That's another option, though given the original request, building with only GTK+2 then becomes:
```
brew install --without-gtk+3 --with-gtk+ gtk-mac-integration
```

**Reply by gromgit** [reply]
@easior Forgot to mention: If you're using `:optional`, you need to read https://docs.brew.sh/Formula-Cookbook#adding-optional-steps as well. Depending on the formula, you may need to add conditional logic to configure the build accordingly.
